spec/JustNotSorrySpec.js

```diff
@@ -91,4 +147,99 @@ describe('JustNotSorry', function () {
       expect(callCount).toEqual(3);
     });
   });
+
+  describe('contentEditable observer', function () {
+    var newDivId = 'div-5';
+
+    afterEach(function () {
+      var targetDiv = document.getElementById(newDivId);
+      targetDiv.parentNode.removeChild(targetDiv);
+    });
+
+    it('dispatches an input event when nodes are added to a content editable div', function (done) {
+      generateEditableDiv(newDivId);
+      const target = document.getElementById(newDivId);
+
+      spyOn(warningChecker, 'addWarnings');
+      spyOn(warningChecker, 'removeWarnings');
+
+      // trigger documentObserver to register this content editable div
+      target.appendChild(document.createElement('BR'));
+
+      setTimeout(function () {
+        dispatchEventOnElement(target, 'focus');
+        expect(warningChecker.addWarnings).toHaveBeenCalledTimes(1);
+        warningChecker.addWarnings.calls.reset();
+
+        setTimeout(function () {
+          let element = document.createElement('SPAN');
+          element.textContent = 'Hello';
+          target.appendChild(element);
+
+          setTimeout(function () {
```

Comment:
These nested timeouts look like they must have been a pain to work with, but the tests all pass

Reply:
They were definitely a pain.  I tried using jasmine's mock clock instead, but it didn't work.  From what I could tell from stack overflow posts, jasmine doesn't mock out enough to be able to trigger the mutation observers. setTimeout was the best I could do.  I'm open to suggestions for better ways to do it.  I'm also hoping that if we convert to jest, we'll be able to rewrite these tests to get rid of the setTimeouts.